src/Common/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,33 @@
+import { useCallback, useState, cloneElement } from 'react'
+import TippyJS from '@tippyjs/react'
+import { TooltipProps } from './types'
+
+const Tooltip = ({
+    showOnTruncate = false,
+    alwaysShowTippyOnHover = false,
+    children: child,
+    ...rest
+}: TooltipProps) => {
+    const [isTextTruncated, setIsTextTruncated] = useState(false)
+
+    const refCallback = useCallback((node: HTMLDivElement) => {
+        if (node) {
+            setIsTextTruncated(node.scrollWidth > node.clientWidth || node.scrollHeight > node.clientHeight)
```

Comment:
For `dc__truncate` we need to check scrollHeight instead. Since it uses line clamp instead of text-overflow

Reply:
we can add in code comment